tf_big/cc/kernels/big_kernels.cc

```diff
@@ -0,0 +1,95 @@
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/shape_inference.h"
+#include "tensorflow/core/framework/variant.h"
+#include "tensorflow/core/framework/variant_encode_decode.h"
+#include "tensorflow/core/framework/variant_op_registry.h"
+#include "tensorflow/core/framework/variant_tensor_data.h"
+#include "tensorflow/core/kernels/bounds_check.h"
+
+#include "big_tensor.h"
+#include "gmp.h"
+
+using namespace tensorflow;
+
+Status GetBigTensor(OpKernelContext* ctx, int index, const BigTensor** res) {
+  const Tensor& input = ctx->input(index);
+
+  // TODO: check scalar type
+  const BigTensor* big = input.scalar<Variant>()().get<BigTensor>();
+  if (big == nullptr) {
+    return errors::InvalidArgument("Input handle is not a mpz wrapper. Saw: '",
+                                   input.scalar<Variant>()().DebugString(),
+                                   "'");
+  }
+
+  *res = big;
+  return Status::OK();
+}
+
+class BigImportOp : public OpKernel {
+ public:
+  explicit BigImportOp(OpKernelConstruction* context) : OpKernel(context) {}
+
+  void Compute(OpKernelContext* ctx) override {
+    const Tensor& str = ctx->input(0);
+    OP_REQUIRES(
+        ctx, TensorShapeUtils::IsMatrix(str.shape()),
+        errors::InvalidArgument("value expected to be a matrix ",
+                                "but got shape: ", str.shape().DebugString()));
+
+    Tensor* val;
+    OP_REQUIRES_OK(ctx, ctx->allocate_output(0, TensorShape{}, &val));
+
+    BigTensor big;
+    big.FromTensor<string>(str);
```

Comment:
we should probably generalise this so you can import from different types

Reply:
just did this in another PR https://github.com/tf-encrypted/tf-big/pull/4

Reply:
only to int32 though (see PR) I think this is sufficient for now